src/components/application_manager/src/rpc_handler_impl.cc

```diff
@@ -118,16 +118,17 @@ void RPCHandlerImpl::ProcessMessageFromMobile(
         (*so_from_mobile)[strings::params][strings::correlation_id].asUInt();
     int32_t message_type =
         (*so_from_mobile)[strings::params][strings::message_type].asInt();
-    if (app_manager_.GetAppServiceManager()
-            .GetRPCPassingHandler()
-            .RPCPassThrough(*so_from_mobile)) {
+    RPCPassingHandler& handler =
+        app_manager_.GetAppServiceManager().GetRPCPassingHandler();
+    // Check permissions for requests, otherwise attempt passthrough
+    if ((application_manager::MessageType::kRequest != message_type ||
+         handler.IsPassthroughAllowed(*so_from_mobile)) &&
+        handler.RPCPassThrough(*so_from_mobile)) {
```

Comment:
@jacobkeeler   I would not keep `handler.IsPassthroughAllowed` and `handler.RPCPassThrough` in one if condition. 
handler.IsPassthroughAllowed - Checking method, it does not have any actions.
handler.RPCPassThrough - Action method. I would avoid keeping this method in if condition, especially within checking method. 
Just to improve readability.